diff --git a/server/mapping.c b/server/mapping.c
index a53b6b68304..ac1f36985a6 100644
--- a/server/mapping.c
+++ b/server/mapping.c
@@ -276,7 +276,7 @@ void init_memory(void)
 {
     host_page_mask = sysconf( _SC_PAGESIZE ) - 1;
     free_map_addr( 0x60000000, 0x1c000000 );
-    free_map_addr( 0x600000000000, 0x100000000000 );
+    free_map_addr( 0x4000000000, 0x1000000000 );
 }
 
 static void ranges_dump( struct object *obj, int verbose )
