From 510defb0fdbec725410165bdecc6710b8b173264 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 5 Dec 2025 11:06:10 +0100
Subject: [PATCH] mfsrcsnk: Factor out a media_source_request_stream_sample
 helper.

---
 dlls/mfsrcsnk/media_source.c | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/dlls/mfsrcsnk/media_source.c b/dlls/mfsrcsnk/media_source.c
index d31be5c3..895b8041 100644
--- a/dlls/mfsrcsnk/media_source.c
+++ b/dlls/mfsrcsnk/media_source.c
@@ -826,25 +826,34 @@ static HRESULT WINAPI media_stream_GetStreamDescriptor(IMFMediaStream* iface, IM
     return hr;
 }
 
-static HRESULT media_stream_async_request(struct media_stream *stream, IMFAsyncResult *result)
+static HRESULT media_source_request_stream_sample(struct media_source *source, struct media_stream *stream, IUnknown *token)
 {
-    struct media_source *source = media_source_from_IMFMediaSource(stream->source);
-    IUnknown *token = IMFAsyncResult_GetStateNoAddRef(result);
     IMFSample *sample;
     HRESULT hr = S_OK;
 
-    EnterCriticalSection(&source->cs);
-
     if (source->state == SOURCE_SHUTDOWN)
-        hr = MF_E_SHUTDOWN;
-    else if (source->state == SOURCE_RUNNING && SUCCEEDED(hr = object_queue_pop(&stream->samples, (IUnknown **)&sample)))
+        return MF_E_SHUTDOWN;
+
+    if (source->state == SOURCE_RUNNING && SUCCEEDED(hr = object_queue_pop(&stream->samples, (IUnknown **)&sample)))
     {
         media_stream_send_sample(stream, sample, token);
         IMFSample_Release(sample);
+        return S_OK;
     }
-    else if (SUCCEEDED(hr = object_queue_push(&stream->tokens, token)) && source->state == SOURCE_RUNNING)
+
+    if (SUCCEEDED(hr = object_queue_push(&stream->tokens, token)) && source->state == SOURCE_RUNNING)
         queue_media_source_read(source);
+    return hr;
+}
 
+static HRESULT media_stream_async_request(struct media_stream *stream, IMFAsyncResult *result)
+{
+    struct media_source *source = media_source_from_IMFMediaSource(stream->source);
+    IUnknown *token = IMFAsyncResult_GetStateNoAddRef(result);
+    HRESULT hr = S_OK;
+
+    EnterCriticalSection(&source->cs);
+    hr = media_source_request_stream_sample(source, stream, token);
     LeaveCriticalSection(&source->cs);
 
     return hr;
-- 
2.43.0

