Adapted to wine-10.15 by Operit.
From: GloriousEggroll <gloriouseggroll@proton.me>
Subject: [PATCH] rtworkq: Release completed serial work items

diff --git a/dlls/rtworkq/queue.c b/dlls/rtworkq/queue.c
index 144feb9..4745537 100644
--- a/dlls/rtworkq/queue.c
+++ b/dlls/rtworkq/queue.c
@@ -455,6 +455,7 @@ static void CALLBACK serial_queue_finalization_callback(PTP_CALLBACK_INSTANCE in
     LeaveCriticalSection(&queue->cs);
 
     IUnknown_Release(&item->IUnknown_iface);
+    IUnknown_Release(&item->IUnknown_iface);
 }
 
 static HRESULT serial_queue_init(const struct queue_desc *desc, struct queue *queue)
