From 7e70075f7dcc23a8e057798cb2719cdc7c738d38 Mon Sep 17 00:00:00 2001
From: Nello De Gregoris <bluechxindv@gmail.com>
Date: Sat, 20 Jun 2026 01:46:43 +0200
Subject: [PATCH] mfsrcsnk: Use SetCurrentPosition() for absolute byte stream
 seeks.

---
 dlls/mfsrcsnk/media_source.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/dlls/mfsrcsnk/media_source.c b/dlls/mfsrcsnk/media_source.c
index c288383b..28b0d34c 100644
--- a/dlls/mfsrcsnk/media_source.c
+++ b/dlls/mfsrcsnk/media_source.c
@@ -1728,7 +1728,9 @@ static NTSTATUS CDECL media_source_seek_cb( struct winedmo_stream *stream, UINT6
     struct media_source *source = CONTAINING_RECORD(stream, struct media_source, winedmo_stream);
     TRACE("stream %p, pos %p\n", stream, pos);
 
-    if (FAILED(IMFByteStream_Seek(source->stream, msoBegin, *pos, 0, pos)))
+    if (FAILED(IMFByteStream_SetCurrentPosition(source->stream, *pos)))
+        return STATUS_UNSUCCESSFUL;
+    if (FAILED(IMFByteStream_GetCurrentPosition(source->stream, pos)))
         return STATUS_UNSUCCESSFUL;
     return STATUS_SUCCESS;
 }
-- 
2.43.0

