From: Operit <operit@local>
Subject: tkg-mf-nogl adapt: dlls/winegstreamer/quartz_parser.c

---
--- a/dlls/winegstreamer/quartz_parser.c
+++ b/dlls/winegstreamer/quartz_parser.c
@@ -1979,6 +1979,16 @@
     /* At least make sure we can convert it to wg_format. */
     return amt_to_wg_format(mt, &format) ? S_OK : S_FALSE;
 }
+static HRESULT decodebin_parser_source_get_media_type(struct parser_source *pin,
+        unsigned int index, AM_MEDIA_TYPE *mt)
+{
+    struct wg_format format;
+
+    TRACE("pin %p, index %d, mt %p.\n", pin, (int)index, mt);
+
+    wg_parser_stream_get_current_format(pin->wg_stream, &format);
+    return get_raw_media_type(&format, index, mt);
+}
 
 static HRESULT get_raw_media_type(struct wg_format *format, unsigned int index, AM_MEDIA_TYPE *mt)
 {
@@ -2716,6 +2726,8 @@
     HRESULT hr;
 
     wg_parser_stream_get_current_format(pin->wg_stream, &format);
+    if (format.major_type == WG_MAJOR_TYPE_VIDEO || format.major_type == WG_MAJOR_TYPE_AUDIO)
+        return amt_to_wg_format(mt, &format) ? S_OK : S_FALSE;
     if (wg_video_format_is_rgb(format.u.video.format))
         format.u.video.height = -format.u.video.height;
     if (!amt_from_wg_format(&pad_mt, &format, false))
@@ -2883,6 +2895,8 @@
     HRESULT hr;
 
     wg_parser_stream_get_current_format(pin->wg_stream, &format);
+    if (format.major_type == WG_MAJOR_TYPE_VIDEO || format.major_type == WG_MAJOR_TYPE_AUDIO)
+        return amt_to_wg_format(mt, &format) ? S_OK : S_FALSE;
     if (!amt_from_wg_format(&pad_mt, &format, false))
         return E_OUTOFMEMORY;
     hr = compare_media_types(mt, &pad_mt) ? S_OK : S_FALSE;
-- 
2.49.0
