From: Operit <operit@local>
Subject: tkg-mf-nogl adapt: dlls/winegstreamer/wg_parser.c

---
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -36,7 +36,6 @@
 #include <gst/audio/audio.h>
 #include <gst/tag/tag.h>
 
-#include <gst/gl/gl.h>
 
 #include "ntstatus.h"
 #define WIN32_NO_STATUS
@@ -45,7 +44,6 @@
 
 #include "unix_private.h"
 
-extern GstGLDisplay *gl_display;
 
 typedef enum
 {
@@ -105,7 +103,6 @@
     struct input_cache_chunk input_cache_chunks[4];
 
     bool use_mediaconv;
-    bool use_opengl;
     GstContext *context;
 };
 static const unsigned int input_cache_chunk_size = 512 << 10;
@@ -1034,36 +1031,18 @@
     name = gst_structure_get_name(gst_caps_get_structure(caps, 0));
     gst_caps_unref(caps);
 
-    if (!strcmp(name, "video/x-raw") && parser->use_opengl)
-    {
-        if (!(element = create_element("glupload", "base"))
-                || !append_element(parser->container, element, &first, &last))
-            return false;
-        if (!(element = create_element("glcolorconvert", "base"))
-                || !append_element(parser->container, element, &first, &last))
-            return false;
-        if (!(element = create_element("glvideoflip", "base"))
-                || !append_element(parser->container, element, &first, &last))
-            return false;
-        stream->flip = element;
-        if (!(element = create_element("gldeinterlace", "base"))
-                || !append_element(parser->container, element, &first, &last))
-            return false;
-        if (!(element = create_element("glcolorconvert", "base"))
-                || !append_element(parser->container, element, &first, &last))
-            return false;
-        if (!(element = create_element("gldownload", "base"))
-                || !append_element(parser->container, element, &first, &last))
-            return false;
-
-        if (!link_src_to_element(pad, first) || !link_element_to_sink(last, stream->my_sink))
-            return false;
-    }
-    else if (!strcmp(name, "video/x-raw"))
+    if (!strcmp(name, "video/x-raw"))
     {
         /* decodebin doesn't provide framerate for raw video. This causes the
          * the deinterlace element to reject the caps. So we need to go through
          * videoconvert first (as it fixates the framerate) */
+
+        /* Hack: Flatten colorimetry to defaults (Proton media-converter). */
+        if (!(element = create_element("capssetter", "good"))
+                || !append_element(parser->container, element, &first, &last))
+            return false;
+        gst_util_set_object_arg(G_OBJECT(element), "join", "true");
+        gst_util_set_object_arg(G_OBJECT(element), "caps", "video/x-raw,colorimetry=0:0:0:0");
 
         /* decodebin considers many YUV formats to be "raw", but some quartz
          * filters can't handle those. Also, videoflip can't handle all "raw"
@@ -2096,16 +2075,6 @@
 
     if (!(parser = calloc(1, sizeof(*parser))))
         return E_OUTOFMEMORY;
-    if ((parser->use_opengl = params->use_opengl && gl_display))
-    {
-        if ((parser->context = gst_context_new(GST_GL_DISPLAY_CONTEXT_TYPE, false)))
-            gst_context_set_gl_display(parser->context, gl_display);
-        else
-        {
-            GST_ERROR("Failed to create parser context");
-            parser->use_opengl = FALSE;
-        }
-    }
     if (!(parser->task_pool = wg_task_pool_new()))
     {
         free(parser);
-- 
2.49.0
