From: Operit <operit@local>
Subject: tkg-mf-nogl adapt: dlls/winegstreamer/unixlib.c

---
--- a/dlls/winegstreamer/unixlib.c
+++ b/dlls/winegstreamer/unixlib.c
@@ -30,7 +30,6 @@
 
 #define GLIB_VERSION_MIN_REQUIRED GLIB_VERSION_2_30
 #include <gst/gst.h>
-#include <gst/gl/gl.h>
 
 #include "ntstatus.h"
 #define WIN32_NO_STATUS
@@ -47,7 +46,6 @@
 GST_DEBUG_CATEGORY(wine);
 
 static UINT thread_count;
-GstGLDisplay *gl_display;
 
 GstStreamType stream_type_from_caps(GstCaps *caps)
 {
@@ -259,7 +257,6 @@
 
 NTSTATUS wg_init_gstreamer(void *arg)
 {
-    static GstGLContext *gl_context;
 
     struct wg_init_gstreamer_params *params = arg;
     char arg0[] = "wine";
@@ -322,28 +319,6 @@
     GST_INFO("GStreamer library version %s; wine built with %d.%d.%d.",
             gst_version_string(), GST_VERSION_MAJOR, GST_VERSION_MINOR, GST_VERSION_MICRO);
 
-    if (!(gl_display = gst_gl_display_new()))
-        GST_ERROR("Failed to create OpenGL display");
-    else
-    {
-        GError *error = NULL;
-        gboolean ret;
-
-        GST_OBJECT_LOCK(gl_display);
-        ret = gst_gl_display_create_context(gl_display, NULL, &gl_context, &error);
-        GST_OBJECT_UNLOCK(gl_display);
-        g_clear_error(&error);
-
-        if (ret)
-            gst_gl_display_add_context(gl_display, gl_context);
-        else
-        {
-            GST_ERROR("Failed to create OpenGL context");
-            gst_object_unref(gl_display);
-            gl_display = NULL;
-        }
-    }
-
     if (!media_converter_init())
     {
         GST_ERROR("Failed to init media converter.");
-- 
2.49.0
